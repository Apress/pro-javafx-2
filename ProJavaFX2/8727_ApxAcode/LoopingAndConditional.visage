/*
 * @test
 * @run
 */

// while loop
var i = 1;
while (i <= 10) {
  println("{i} squared: {i * i}");
  i += 1;
}

// break from while loop
var j = 1;
while (j <= 10) {
  if (j mod 2 != 0) {
    j += 1;
    continue;
  } else {
    println(j);
    j += 1;
  }
}

// continue in for loop
for (x in [1..10]) {
  if (x > 7) {
    break;
  } else {
    println(x);
  }
}

// if expressions
var k = if (i == j) then 1 else 0;

if (k < 0) {
  println("{k} < 0");
} else {
  println("{k} >= 0");
}

for (x in [-1..1]) {
  if (x < 0) {
    println("{x} < 0");
  } else if (x == 0) {
    println("{x} = 0");
  } else {
    println("{x} > 0");
  }
}

// if expression with dissimilar then and else clauses
var a = if (true) then 3 else "4";
println("a = {a}");
// assign an Integer to a
a = 5;
println("a = {a}");
// assign a Stirng to a
a = "hi";
println("a = {a}");

var b = if (true) then 7 else [8];
print("b = "); println(b);
