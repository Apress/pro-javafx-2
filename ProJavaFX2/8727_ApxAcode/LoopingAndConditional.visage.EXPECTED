1 squared: 1
2 squared: 4
3 squared: 9
4 squared: 16
5 squared: 25
6 squared: 36
7 squared: 49
8 squared: 64
9 squared: 81
10 squared: 100
2
4
6
8
10
1
2
3
4
5
6
7
1 >= 0
-1 < 0
0 = 0
1 > 0
a = 3
a = 5
a = hi
b = [ 7 ]
